store/tikv/lock_resolver.go

```diff
@@ -242,11 +242,16 @@ func (lr *LockResolver) BatchResolveLocks(bo *Backoffer, locks []*Lock, loc Regi
 		// Then we need to check the secondary locks to determine the final status of the transaction.
 		if status.primaryLock != nil && status.primaryLock.UseAsyncCommit {
 			resolveData, err := lr.checkAllSecondaries(bo, l, &status)
-			if err != nil {
-				return false, err
+			if err == nil {
+				txnInfos[l.TxnID] = resolveData.commitTs
+				continue
+			}
+			if _, ok := err.(*nonAsyncCommitLock); ok {
+				status, err = lr.getTxnStatus(bo, l.TxnID, l.Primary, callerStartTS, math.MaxUint64, true, true)
+				if err != nil {
+					return false, err
+				}
 			}
```

Comment:
I think you should return the error if the err is not nonAsyncCommitLock here, to be consistent with old logic.

Reply:
oh, sorry...